From 1df97c1f74db654a71b97f05e4b17cda48cfbbfc Mon Sep 17 00:00:00 2001
From: Fei Yang <fei.yang@intel.com>
Date: Sun, 29 Jun 2014 22:23:28 -0700
Subject: [PATCH] EXTCON: TSU6111: link list corruption

The probe function needs to destroy the wake lock in case
of failure. Otherwise the wakeup_sources link list is corrupted.

Change-Id: I37cbea21aadf35126f1a9b989bcbaf57ef6f8eba
Signed-off-by: Fei Yang <fei.yang@intel.com>
---
 drivers/extcon/extcon-tsu6111.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/extcon/extcon-tsu6111.c b/drivers/extcon/extcon-tsu6111.c
index 0451ad9..7e3eb39 100644
--- a/drivers/extcon/extcon-tsu6111.c
+++ b/drivers/extcon/extcon-tsu6111.c
@@ -560,6 +560,7 @@ otg_reg_failed:
 extcon_reg_failed:
 	kfree(chip->edev);
 extcon_mem_failed:
+	wake_lock_destroy(&chip->wakelock);
 	kfree(chip);
 	return ret;
 }
@@ -571,6 +572,7 @@ static int tsu6111_remove(struct i2c_client *client)
 	free_irq(client->irq, chip);
 	usb_put_phy(chip->otg);
 	extcon_dev_unregister(chip->edev);
+	wake_lock_destroy(&chip->wakelock);
 	kfree(chip->edev);
 	pm_runtime_get_noresume(&chip->client->dev);
 	kfree(chip);
-- 
2.0.0

